zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -721,6 +721,10 @@ private void setupMyId() throws IOException {
             throw new IllegalArgumentException("serverid " + myIdString
                     + " is not a number");
         }
+        //myid must be in [0, 255]
+        if (serverId < 0 || serverId > 255) {
```

Comment:
I believe the recent changes for TTL limit this to 254?

@Randgalt can you weigh in here.

Reply:
Yes - this is that whole can of worms fixed in ZOOKEEPER-2901. The max ServerId is determined by whether or not TTL nodes are enabled via `zookeeper.extendedTypesEnabled` - the docs now stipulate: `IMPORTANT: if you enable extended features such as TTL Nodes ... the id must be between 1 and 254 due to internal limitations.`